### ISSUE TITLE
"[...]/node_modules/@types/element-resize-detector/index" has no default export.

### ISSUE THREAD

**Issue by sefn** [sebgroup/ng-magic-iframe#6]
When importing ng-magic-iframe into another angular repo and trying to build:

`ERROR in node_modules/@sebgroup/ng-magic-iframe/lib/components/ng-magic-iframe.component.d.ts(3,8): error TS1192: Module '"C:/ws/git/arena-stefan-example/shell-app/node_modules/@types/element-resize-detector/index"' has no default export.`

app.module.ts:

```
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { NgMagicIframeModule } from '@sebgroup/ng-magic-iframe';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    NgMagicIframeModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }

```

**Comment by sefn** [comment]
@hjalmers  I think it's this import statement missing curly braces:

https://github.com/sebgroup/ng-magic-iframe/blob/a27cada9e7aa0273e0813d08f930348816bb5bbc/projects/ng-magic-iframe/src/lib/components/ng-magic-iframe.component.ts#L13

**Comment by sefn** [comment]
@hjalmers Created PR https://github.com/sebgroup/ng-magic-iframe/pull/7 (but can't set you as reviewer)

**Comment by sefn** [comment]
I fixed it in my project that uses ng-magic-iframe by adding this to my tsconfig.json:

```
"allowSyntheticDefaultImports": true,
```

I think it's probably something with how tsc transpiles the element-resize-detector. This is my complete tsconfig.json:

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ]
  }
}
```

And this is how the .\node_modules\@types\element-resize-detector\index.d.ts looks in my angular project that npm installed ng-magic-iframe:

```
// Type definitions for element-resize-detector 1.1
// Project: https://github.com/wnr/element-resize-detector
// Definitions by: Saransh Kataria <https://github.com/saranshkataria>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

declare function elementResizeDetectorMaker(options?: elementResizeDetectorMaker.ErdmOptions): elementResizeDetectorMaker.Erd;

declare namespace elementResizeDetectorMaker {
    interface ErdmOptions {
    strategy?: 'scroll' | 'object';
    }

    interface Erd {
    listenTo(element: HTMLElement, callback: (elem: HTMLElement) => void): void;
    removeListener(element: HTMLElement, callback: (elem: HTMLElement) => void): void;
    removeAllListeners(element: HTMLElement): void;
    uninstall(element: HTMLElement): void;
    }
}
export = elementResizeDetectorMaker;
```

**Comment by hjalmers** [comment]
I've added the instructions to the readme. 

I thought about updating the typings definition for `element-resize-detector` but as they're currently `correct` when not using ES6 I passed on that idea as switching from `export = elementResizeDetectorMaker;` to `export default elementResizeDetectorMaker;` would break the build for everyone not using ES6. Adding `allowSyntheticDefaultImports` seems to be the appropriate way to allow for backwards compatibility.
